plugin/parser/ast/table_def.go

```diff
@@ -1,9 +1,23 @@
 package ast
 
+// TableType is the type for table.
+type TableType int
+
+const (
+	// TableTypeNone is the type for table or view, which means we can't be sure of it.
+	TableTypeNone TableType = iota
```

Comment:
As a 2nd thought. I feel TableTypeUnknown is better:

1. For the index case, the table does have type, the parser just does not have enough context to know.

2. For the convert case, it's for returning the error, thus Unknown is a common pattern.

wdyt?

Reply:
SGTM
Naming is worthy of the hardest problem in software engineering.

Reply:
From API, it's usually called XxUnspecified or UnknownXx.

https://google.aip.dev/126